modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/stored/PutStoredTemplateRequest.java

```diff
@@ -0,0 +1,124 @@
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class PutStoredTemplateRequest extends AcknowledgedRequest<
+        PutStoredTemplateRequest> {
+    private String id;
+    private BytesReference content;
+    private XContentType xContentType;
+
+    public PutStoredTemplateRequest() {
+    }
+
+    public PutStoredTemplateRequest(String id, BytesReference content,
+            XContentType xContentType) {
+        this.id = id;
+        this.content = content;
+        this.xContentType = Objects.requireNonNull(xContentType);
+    }
+
+    public PutStoredTemplateRequest id(String id) {
+        this.id = id;
+        return this;
+    }
+
+    public PutStoredTemplateRequest content(BytesReference content,
+            XContentType xContentType) {
+        this.content = content;
+        this.xContentType = xContentType;
+        return this;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+
+        if (id == null || id.isEmpty()) {
+            validationException = addValidationError("must specify id for stored search template",
+                    validationException);
+        } else
+            if (id.contains("#")) {
+                validationException = addValidationError(
+                        "id cannot contain '#' for stored search template", validationException);
+            }
+
+        if (content == null) {
+            validationException = addValidationError("must specify code for stored search template",
+                    validationException);
+        }
+
+        return validationException;
+    }
+
+    public String id() {
+        return id;
+    }
+
+    public BytesReference content() {
+        return content;
+    }
+
+    public XContentType xContentType() {
+        return xContentType;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        super.readFrom(in);
+        id = in.readOptionalString();
+        content = in.readBytesReference();
+        xContentType = XContentType.readFrom(in);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeOptionalString(id);
+        out.writeBytesReference(content);
+        xContentType.writeTo(out);
+    }
+
+    @Override
+    public String toString() {
+        String source = "_na_";
+
+        try {
+            source = XContentHelper.convertToJson(content, false, xContentType);
+        } catch (Exception e) {
+            // ignore
+        }
+
+        return "put search template {id [" + id + "], content [" + source + "]}";
```

Comment:
Same comment here about following the same pattern for these (regardless of what the pattern ends up being)